### ISSUE TITLE
The strange camera inputs

### ISSUE THREAD

**Issue by zhanghm1995** [VAIL-UCLA/BridgeSim#18]
Hi, thanks for sharing this wonderful work. 

When I attempt to visualize the image inputs of the model (`cam_feat = model_input['camera_feature']`), I obtain the strange camera order:

It seems the camera left and camera right are opposite. Is this the expected behavior?

https://github.com/user-attachments/assets/b559b916-cf6a-4625-94da-c75e6e6cb092

I used the following code in the `bridgesim/evaluation/core/base_evaluator.py` to save the model inputs:
```python
self.render_cam_f0_vis(env, frame_id, ego_state, plan_traj)

if 'camera_feature' in model_input:
    cam_feat = model_input['camera_feature']
    cam_np = cam_feat[0].cpu().numpy() if hasattr(cam_feat, 'cpu') else cam_feat[0]
    cam_rgb = (cam_np.transpose(1, 2, 0) * 255).clip(0, 255).astype('uint8')
    cam_bgr = cam_rgb[:, :, ::-1]
    frame_dir = self.output_dir / f"{frame_id:05d}"
    frame_dir.mkdir(parents=True, exist_ok=True)
    cv2.imwrite(str(frame_dir / "model_input.jpg"), cam_bgr)
```

BTW, could we speed up the evaluation process? It needs almost more than 3 hours in headless H100 server to ran the evaluation on the 421 NavHard split. And when I try to start up multiple processes, it would freeze.

**Comment by RobinWangSD** [comment]
Thanks for pointing out the issue. We recently updated our camera setting, which may have caused this. We'll investigate and fix any potential bugs.

Unfortunately, our current setup does not support acceleration for training or evaluation. We are actually working on a new parallelizable simulator that supports pixel-level rendering and generation. Please stay tuned!

**Comment by zhanghm1995** [comment]
When I fixed this camera order issue, I found the DS could be improved to some extent. And I also found the pretrained policy in NAVSIM uses the RGB channels order. But MetaDrive uses the BGR channels. When I fixed these two problems (channels order & camera order), I got a 1-point increase. 

I'm glad to hear that you are working on a new parallelizable simulator. Hope this new simulator will be released soon; it would accelerate the development of closed-loop simulators for autonomous driving.

**Comment by RobinWangSD** [comment]
Thanks for pointing out the issue—we've found the cause. In our internal code, the camera configuration is set up per model based on each model's source dataset. When we cleaned and open-sourced the code, we merged these configurations together, which caused the issue. 

We'll also check and fix any potential color channel issues introduced during the merge and cleanup. In our local code, the color channel is also handled differently for each model adaptor.

Our reported scores were obtained in the internal codebase. We would expect the scores after the fix to be similar to the reported ones (a ~1 point increase would fall within this range).

**Comment by zhanghm1995** [comment]
I'm glad you found the issue. Could you align the public source code as closely as possible with your local version? I believe this will help the community get the code up and running with minimal issues. Thanks!

**Comment by RobinWangSD** [comment]
We agree, and aligning the public code with our local version is exactly our goal. The challenge is that the local branch contains a lot of duplicate and experiment-specific code, so a direct copy would be hard for the community to use. We will continue reconciling the public branch with the internal version. For issues like this, we trace the source of the discrepancy, identify whether it was introduced during cleanup, and patch the public branch accordingly. Please keep reporting any further inconsistencies you find, as this helps us close the gap quickly. Thanks again for the helpful feedback.

**Comment by HongweiRuan** [comment]
Hi @zhanghm1995 , we have pushed code to fix the problem of left right camera and RGB channels, thank you for pointing this out!

**Comment by zhanghm1995** [comment]
Hi, thanks for fixing this problem. I pulled the latest code and tested it again. The camera order seems correct now. I also noticed that the trajectory path (wide cyan line), the target goal point (green dot), and the future ground truth trajectory (red dotted line) appear abnormal. They are not consistent with the corresponding elements in the top-down view.

The scene: `sd_352290f8ed8e572e`

<img width="800" height="800" alt="Image" src="https://github.com/user-attachments/assets/096c67af-9ee7-4f4d-8a0e-56ba6ba24b82" />

https://github.com/user-attachments/assets/36987d01-424f-4bec-9e64-fcd818f72ca9

Testing command:
```bash
python bridgesim/evaluation/unified_evaluator.py \
      --model-type transfuser \
      --checkpoint  ckpts/BridgeSim/navsimv2/transfuser/transfuser_seed_0.ckpt \
      --scenario-path "dataset/bridgesim_navhard_converted/$scene_name" \
      --output-dir outputs/visualize \
      --traffic-mode log_replay --eval-mode closed_loop \
      --controller pure_pursuit --replan-rate 5 \
      --ego-replay-frames 20 --eval-frames 80 \
      --enable-vis  --no-save-perframe
```

**Comment by HongweiRuan** [comment]
Hi @zhanghm1995, thanks again for pointing out another issue, we have fixed the camera overlay intrinsics, you can try again with the latest code!

**Comment by zhanghm1995** [comment]
Thanks for fixing this new problem. I pulled the latest code and tested the same scene, I got the following result. I'm not sure whether the green line is correct.

https://github.com/user-attachments/assets/817f5332-89fe-4ee9-a05d-c68e82fe27d4

**Comment by HongweiRuan** [comment]
Hi @zhanghm1995, the green vertical pole is used to track the expert trajectory, it's an intentional design.

**Comment by zhanghm1995** [comment]
But why is the expert trajectory not smooth?  BTW, why are the trucks on the left road jumping up and down?

**Comment by RobinWangSD** [comment]
The expert trajectory is shown by the red waypoints. The green line is a vertical stick placed at the first upcoming waypoint of the expert trajectory. It's used to compute the action category (left, straight, right) on the fly.

The stacked vehicle is a carryover bug from MetaDrive + ScenarioNet. The recorded bounding box labels are noisy—type and/or size can vary between time frames—which conflicts with the physics simulation. However, we don't currently have a plan to fix this in this version.
